### PR TITLE
refactor: update GradebookFilter components to function component

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -** refactor component to use hooks

JIRA: [AU-935](https://2u-internal.atlassian.net/browse/AU-935)

**What changed?**
- Incremental component updates to functional components.
- Moving redux selectors/actions/thunkActions into hooks as we go.
- Unit test to 100%
image
- coverage test badge
image

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [x] Received code-owner approving review
- [x] Bumped version number [package.json](../package.json)

**Testing Instructions**

- Run the application and test filtering functionality

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by nsprenkle** [comment]
**nit** technically this PR should be under the `refactor` conventional commit tag.

**Comment by nsprenkle** [comment]
**second nit** and the PR template is not filled out. Please update at earliest convenience.

**Review by schenedx — APPROVED** [review]
👍 Let's do this
